frappe/public/js/frappe/form/controls/attach.js

```diff
@@ -32,6 +32,13 @@ frappe.ui.form.ControlAttach = class ControlAttach extends frappe.ui.form.Contro
 
 		frappe.utils.bind_actions_with_object(this.$value, this);
 		this.toggle_reload_button();
+
+		// Don't allow attaching to child tables for new documents
+		if (this.frm.doc.__islocal && this.doc.parenttype == this.frm.doc.doctype) {
```

Comment:
`frm` won't exist on control used outside of form.

Reply:
ah right